src/BSeries.jl

```diff
@@ -1453,4 +1456,379 @@ include("latexify.jl")
+        order = order + 1
+    end
+    return energy_preserving_flag
+end
+
+#       low_order_energy_preserving(trees,coefficients)
+#
+# This function checks whether the set of level_sequences 'trees'
+# and the set of 'coefficients' corresponding to a Bseries is 
+# energy_preserving up to the 'uppermost_order', which we choose 
+# to be 4 for computational optimization
+function low_order_energy_preserving(trees, coefficients)
+    #set the limit up to which this function will work
+    uppermost_order = 4
+    same_order_trees = []
+    same_order_coeffs = []
+    index = 1
+    while length(trees[index]) <= uppermost_order
+        push!(same_order_coeffs, Rational{Int64}(coefficients[index]))
+        push!(same_order_trees,trees[index])
+        index += 1
+        if index > length(trees)
+            break
+        end
+    end
+    return energy_preserving_trees_test(same_order_trees,same_order_coeffs)
+end
+
+#        remove_trunk(a)
+
+#This function returns the forest of level_sequences 
+#obtained after removing the rightmost trunk of a given
+#level_sequence.
+function remove_trunk(a)
+    m = num_branchs(a)
+    branchs_array = Array{Array}(undef, m)
+    #we need to save the final number in the level_sequence because this is the final branch of the trunk
+    k = a[end]
+    #we need to look for the last last_j_occurrence of every integer in [1,k-1]
+    for j in 1:k-1
+        last_j_occurrence = findlast(x -> x == j, a)
+        last_jplus1_occurrence = findlast(x -> x == j+1, a)
+        #consider the empty branchs
+        if isnothing(last_j_occurrence) || isnothing(last_jplus1_occurrence)
+            branchs_array[j] = []
+        else
+            branchs_array[j] = a[last_j_occurrence+1:last_jplus1_occurrence-1]
+        end
+    end
+    return branchs_array
+end
+
+
+#        get_branchs()
+
+#It is not enough to generate the branchs and swap them. The level_sequences must be modified 
+#with corrections to the numbers inside: the numbers will decrease if the branch is moved to a 
+#lower position, and they will increase if they are relocated in an upper position.
+
+function get_branchs(a)
+    #we obtain the branchs via 'remove_trunk'
+    #save them in 'branchs_array'
+    branchs_array = remove_trunk(a)
+    m = num_branchs(a)
+    #create another dict for the modified indexes
+    branchs = Array{Array}(undef, m)
+    mid_branch = (m+1)/2
+    #we check if the number of branchs is odd:
+    #in that case, the middle one remains the same
+    for j in 1:m
+        if m % 2 == 1 && j == mid_branch
+            branchs[j] = branchs_array[j]
+            #we use the formula
+            #n+m-2j+1 for every number in the level_sequence
+        else
+            branchs[j] = [n+m-2j+1 for n in branchs_array[j]]
+        end
+    end
+    return reverse(branchs)
+end
+
+
+#This functions checks if the level_sequence is a bush.
```

Comment:
Should be "This function" (no "s" at the end).

Reply:
Implemented.